nucypher/cli/characters/enrico.py

```diff
@@ -7,73 +9,81 @@
 from nucypher.cli.types import NETWORK_PORT
 
 
-@click.command()
-@click.argument('action')
+# Args (policy_encrypting_key)
+def _common_options(func):
+    @click.option('--policy-encrypting-key', help="Encrypting Public Key for Policy as hexadecimal string", type=click.STRING, required=True)
+    @functools.wraps(func)
+    def wrapper(*args, **kwargs):
+        return func(*args, **kwargs)
+    return wrapper
+
+@click.group()
+def enrico():
+    """
+    Enrico the Encryptor" management commands.
+    """
+    pass
+
+
+@enrico.command()
+@_common_options
 @click.option('--dry-run', '-x', help="Execute normally without actually starting the node", is_flag=True)
 @click.option('--http-port', help="The host port to run Moe HTTP services on", type=NETWORK_PORT)
-@click.option('--message', help="A unicode message to encrypt for a policy", type=click.STRING)
-@click.option('--policy-encrypting-key', help="Encrypting Public Key for Policy as hexidecimal string", type=click.STRING)
 @nucypher_click_config
-def enrico(click_config, action, policy_encrypting_key, dry_run, http_port, message):
+def run(click_config,
+
+        # Common
+        policy_encrypting_key,
+
+        # Other
+        dry_run, http_port):
+    """
+    Start Enrico's controller.
     """
-    "Enrico the Encryptor" management commands.
+    ENRICO = _create_enrico(click_config, policy_encrypting_key)
 
-    \b
-    Actions
-    -------------------------------------------------
-    \b
-    run       Start Enrico's controller.
-    encrypt   Encrypt a message under a given policy public key
+    # RPC
+    if click_config.json_ipc:
+        rpc_controller = ENRICO.make_rpc_controller()
+        _transport = rpc_controller.make_control_transport()
+        rpc_controller.start()
+        return
 
+    ENRICO.log.info('Starting HTTP Character Web Controller')
+    controller = ENRICO.make_web_controller()
+    return controller.start(http_port=http_port, dry_run=dry_run)
+
+
+@enrico.command()
+@click.option('--message', help="A unicode message to encrypt for a policy", type=click.STRING, required=True)
+@_common_options
+@nucypher_click_config
+def encrypt(click_config,
+
+            # Other (required)
+            message,
+
+            # Common
+            policy_encrypting_key,
+            ):
+    """
+    Encrypt a message under a given policy public key.
     """
+    ENRICO = _create_enrico(click_config, policy_encrypting_key)
 
-    #
-    # Validate
-    #
+    # Request
+    encryption_request = {'message': message}
+    response = ENRICO.controller.encrypt_message(request=encryption_request)
+    return response
 
-    if not policy_encrypting_key:
-        raise click.BadArgumentUsage('--policy-encrypting-key is required to start Enrico.')
 
-    # Banner
+def _create_enrico(click_config, policy_encrypting_key):
```

Comment:
I think it is better separated into `_setup_emitter()` and `_create_enrico()` proper, similarly to other characters.

Reply:
✔️